contracts/RewardController.sol

```diff
@@ -143,6 +144,9 @@ contract RewardController is IRewardController, OwnableUpgradeable {
         uint256 _sharesChange,
         bool _isDeposit
     ) internal {
+        if (IHATVault(_vault).renouncedRewards(_user)) {
```

Comment:
and then it can also revert with a relevant error. and also it should revert if the user had renounced (otherwise he can still withdraw the rewards he acuumulated before renouncing, which kind of beats the point of renouncing, no?)